modules/shortcodes/vimeo.php

```diff
@@ -259,10 +284,10 @@ function vimeo_link( $content ) {
  * @return string The Vimeo HTML embed code.
  */
 function vimeo_link_callback( $matches ) {
-	if ( isset( $matches[5] ) && ctype_digit( $matches[5] ) ) {
-		return "\n" . vimeo_shortcode( array( 'id' => $matches[5] ) ) . "\n";
+	$id = isset( $matches[ 2 ] ) ? $matches[ 2 ] : $matches[ 1 ];
+	if ( isset( $id ) && ctype_digit( $id ) ) {
+		return "\n" . vimeo_shortcode( array( 'id' => $id ) ) . "\n";
 	}
-	return '';
```

Comment:
why note return `$matches[ 0 ]` here? that would mean that if we have an invalid URL, then we at least leave the text unchanged instead of replacing with with a blank string